test/view_component/view_component_test.rb

```diff
@@ -386,6 +386,84 @@ def test_backtrace_returns_correct_file_and_line_number
     assert_match %r[app/components/exception_in_template_component\.html\.erb:2], error.backtrace[0]
   end
 
+  # Some example usages:
+  #
+  # <%= render(ProductComponent.all(@products)) %>
+  # <%= render(ProductComponent.all(collection: @products, as: :product, foo: foo)) %>
+  # <%= render(ProductComponent.new(product: @product)) %>
```

Comment:
It would be terrific for these tests to be stronger a) to provide deeper "documentation" on how this feature works b) there are tests in here that will pass even if the features are broken.

Reply:
I reworked these tests with the new api and tried to make them do a better job of documenting the feature, let me know what you think. There shouldn't be any that pass even if the feature is broken, can you comment on the specific test if that's still the case?